client/my-sites/email/form/mailboxes/types.ts

```diff
@@ -10,12 +11,42 @@ enum EmailProvider {
 
 interface MailboxFormField< Type > {
 	error: FieldError;
+	isRequired: boolean;
+	isVisible: boolean;
+	readonly typeName: string;
 	value: Type;
 }
 
 abstract class MailboxFormFieldBase< T > implements MailboxFormField< T > {
-	error: FieldError = null;
+	private _error: FieldError = null;
+
+	public get error() {
+		return this._error;
+	}
+
+	public set error( _error: FieldError ) {
```

Comment:
Prefixing this method parameter with an underscore doesn't seem right, I'm not sure why we would want to do that:

```suggestion
	public set error( error: FieldError ) {
```